content/news/2019-12-03-mapstruct-and-quarkus.md

```diff
@@ -0,0 +1,220 @@
+---
+title: "MapStruct and Quarkus - a match made in heaven?"
+author: Christian Bandowski
+date: "2019-12-03"
+tags: [news, examples]
+---
+
+This year is already over, but it was started with something new that came up in the Java world: [Quarkus](https://quarkus.io/). You may already have heard about it, if not, don’t worry, I will quickly summarize what it is.
+
+_tl;dr: Take a look at the README in our [examples repository](https://github.com/mapstruct/mapstruct-examples/tree/master/mapstruct-quarkus)._
+
+<!--more-->
+
+### Quarkus
+> Supersonic Subatomic Java
+> 
+> A Kubernetes Native Java stack tailored for GraalVM & OpenJDK HotSpot, crafted from the best of breed Java libraries and standards
+
+That’s the description on [quarkus.io](https://quarkus.io/), but what does this mean?
+
+With Quarkus you can built microservices or other Java applications using a stack of already existing components like [Eclipse Vert.x](https://vertx.io/), [Hibernate](http://hibernate.org/) or [Eclipse MicroProfile](https://microprofile.io/). It has a container-first philosophy which means that Quarkus sticks all these frameworks together in a way that the resulting application works very well in container environments like [Kubernetes](https://kubernetes.io/) or [OpenShift](https://openshift.io/).
+
+Read [this article](http://in.relation.to/2019/03/08/why-quarkus/) if you want to know more about Quarkus itself.
+
+#### And what is so special when using it with MapStruct?
+The important thing is basically that Quarkus runs on the [GraalVM](https://www.graalvm.org/) _(but also on any other JVM)_ that provides ahead-of-time compilation to create a native image of your application (native system-dependent machine code and not bytecode). Thus you don’t need to have an own JVM installation to run this code. Similar to a regular compiled C++ or Go application. \
+This allows having the advantages of native applications also for JVM based apps like a much faster startup time, less memory usage and a much smaller image (just a few megabytes instead of easily more than 100 MB for the JVM and required libraries).
```

Comment:
JVM-based -> Java-based (as JVM and GraalVM are the both opposing execution modes)

Reply:
That is actually one misconception about GraalVM. GraalVM and JVM are not opposing execution modes. You can use GraalVM only as a JIT compiler (instead of Hotspot C2). GraalVM Native is something extra from the Graal team. Twitter actually uses GraalVM (JIT Mode) in production.